test/org/opensolaris/opengrok/index/IgnoredNamesTest.java

```diff
@@ -71,6 +71,10 @@ public void testIgnoredPatterns() {
         assertFalse(instance.ignore("usr/src/foo/bar/usr.lib/main.c"));
         assertFalse(instance.ignore("usr/src/foo/bar/usr.lib"));
 
+        /* Test handling of special directories. */
+        assertTrue(instance.ignore("usr/src/.git"));
+        assertFalse(instance.ignore("usr/src/foo.git"));
+
```

Comment:
can we add a test where only FILE .repo will be ignored
but path which has .repo in it won't?